src/Google/GlobalSiteTag.php

```diff
@@ -583,4 +648,57 @@ private function register_js_for_fast_refresh_dev() {
 			false
 		);
 	}
+
+	/**
+	 * Get the config object for Google tag.
+	 *
+	 * @return array
+	 */
+	private function get_config_object(): array {
+		// Standard config.
+		$config = [
+			'groups'         => 'GLA',
+			'send_page_view' => false,
+		];
+
+		// Check if enhanced conversion is enabled.
+		if ( $this->is_enhanced_conversion_enabled() ) {
+			$config['allow_enhanced_conversions'] = true;
+		}
+
+		return $config;
+	}
+
+	/**
+	 * Checks if enhanced conversion is enabled.
+	 *
+	 * @return bool
+	 */
+	private function is_enhanced_conversion_enabled(): bool {
+		// Check if enhanced conversion is enabled.
+		$enhanced_conversion_status = $this->options->get( OptionsInterface::ADS_ENHANCED_CONVERSION_STATUS, null );
+		return ( 'enabled' === $enhanced_conversion_status );
+	}
+
+	/**
+	 * Return the hashed data to be sent to Google Ads for enhanced conversion.
+	 *
+	 * @param string $value Data that needs to be hashed.
+	 * @param string $algo  Algorithm for hashing.
+	 *
+	 * @return string
+	 */
+	private function normalize_and_hash( string $value, $algo = 'sha256' ): string {
```

Comment:
Does this function apply all the hashing rules? Looking at this documentation: https://developers.google.com/google-ads/api/docs/conversions/enhanced-conversions/web#php

This one would be missing:
> Remove all periods (.) that precede the domain name in gmail.com and googlemail.com email addresses

Reply:
Added the normalization to remove any periods in the email address if has domain gmail.com or googlemail.com